src/app/Plans/components/helpers.ts

```diff
@@ -1,122 +1,175 @@
+      break;
+    }
+    case state === 'Copying':
+    case state === 'PipelineRunning': {
+      title = isWarmPlan ? 'Running cutover' : 'Running';
+      break;
+    }
+    case state === 'Finished-Incomplete': {
+      title = 'Finished - Incomplete';
+      variant = ProgressVariant.warning;
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
+
+export const getButtonState = (state: PlanState | null): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case state === 'Starting': {
+      type = null;
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'Finished-Incomplete':
+    case state === 'Finished-Failed':
+    case state === 'Canceled':
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null;
+      break;
+    }
+    case state === 'PipelineRunning': {
+      type = null;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      type = null;
+      break;
+    }
+    default: {
+      type = null;
+    }
+  }
+
+  return type;
+};
+
+export const getPlanState = (
   plan: IPlan | null,
-  migration: IMigration | null,
-  migrationsQuery: UseQueryResult<IKubeList<IMigration>>
-): WarmPlanState | null => {
-  if (!plan || !plan.spec.warm) return null;
+  migration: IMigration | null
+): PlanState | null => {
+  if (!plan) return null;
+
   if (!migration) return 'NotStarted';
-  if (isPlanBeingStarted(plan, migration, migrationsQuery)) {
-    return 'Starting';
-  }
+
   const conditions = plan.status?.conditions || [];
+
   if (
-    (hasCondition(conditions, PlanStatusType.Canceled) ||
-      hasCondition(conditions, PlanStatusType.Failed)) &&
+    (hasCondition(conditions, 'Canceled') || hasCondition(conditions, 'Failed')) &&
     !migration.spec.cutover
   ) {
     return 'AbortedCopying';
   }
-  if (!!migration && !!plan.status?.migration?.completed) return 'Finished';
-  if (hasCondition(conditions, PlanStatusType.Executing)) {
+
+  if (!!plan.status?.migration?.started && canPlanBeStarted(plan)) {
+    return 'Finished-Incomplete';
+  }
+
+  if (!!migration && !!plan.status?.migration?.completed) {
+    return 'Finished-Succeeded';
+  }
+
+  if (hasCondition(conditions, 'Executing')) {
     const pipelineHasStarted = plan.status?.migration?.vms?.some((vm) =>
       vm.pipeline.some((step) => !!step.started)
     );
     const cutoverTimePassed =
       migration?.spec.cutover && new Date(migration.spec.cutover).getTime() < new Date().getTime();
+
     if (cutoverTimePassed && !pipelineHasStarted) {
       return 'StartingCutover';
     }
     if (cutoverTimePassed && pipelineHasStarted) {
-      return 'Cutover';
+      return 'PipelineRunning';
```

Comment:
Again, cutover time is a warm migration thing, cold migrations will never have a cutover time and therefore they will never hit this case.

Oh.. but you return PipelineRunning as a base case, so that's okay... yeah I suppose that's okay.

The Copying case also doesn't check if a plan is warm, but it's okay because a cold plan won't have a precopies array... hmm.

I think it might be more clear if you put the 3 warm-specific states (AbortedCopying, StartingCutover, Copying) under an `if(plan.spec.warm) {}` block. Cold plans don't need to bother running that logic and it'll be more readable imo.